### PR TITLE
Added details to the manager, sidecar and SP service sections.

### PR DESCRIPTION
**Pull request by carlbraganza**
Added details to the manager, sidecar and SP service sections. Fixed session figure errors and rewrote the client gRPC description in the risks section.

### CODE REVIEW

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md [hunk 1]
```diff
@@ -787,9 +785,27 @@ The following conditions are well defined:
...
+especially if there is a well defined error message returned.
+
+Thinking ahead to the possibility of a companion data service, then a Capability RPC
+in the SnapshotMetadata service makes sense, because it can be used later to advertise
+the existence of this data service.
+It is also possible that if block v/s extent is fixed for a given driver then that
+could be returned here, though I'm not sure about block size - it may could vary
+by volume.
+<<[/UNRESOLVED]>>
+```
```
Comment (carlbraganza) [inline keps/sig-storage/3314-csi-changed-block-tracking/README.md]: @PrasadG193 @ihcsim Please provide your thoughts.

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md [hunk 2]
```diff
@@ -827,14 +843,143 @@ The following Kubernetes components are involved at runtime:
...
+
+- Is it fixed, say by a ConfigMap the manager could reference?
+  (One size-fits-all-CSI-drivers)
+- Does it depend on the number of snapshots specified in the session?
+- Does it depend on the size of the underlying volumes?
+
+The ConfigMap would be the easiest to implement. The other variants
+would require interaction with the SP.
+<<[/UNRESOLVED]>>
+```
```
Comment (carlbraganza) [inline keps/sig-storage/3314-csi-changed-block-tracking/README.md]: @PrasadG193 @ihcsim Please provide your thoughts.

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md (hunk 1 above)
Comment (ihcsim) [reply keps/sig-storage/3314-csi-changed-block-tracking/README.md]: From the backup application perspective, can we add a "fail" or "unsupported" status to the `SnapshotSessionRequest` if the CSI driver doesn't support this feature? Looking for the `SnapshotServiceConfiguration` is not a bad way - it should be quick to determine if a `SnapshotServiceConfiguration` for a CSI driver exists (via K8s resource labelling). Or we can force the sidecar to register itself with the manager, but it feels like more work with little gain.

I think the capability flag is meant for the CSI plugin to tell the sidecar whether the plugin implements the RPC over the Unix socket.

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md (hunk 2 above)
Comment (ihcsim) [reply keps/sig-storage/3314-csi-changed-block-tracking/README.md]: For alpha, a config map makes sense to me.

**Review by ihcsim — APPROVED** [review]

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md (hunk 1 above)
Comment (PrasadG193) [reply keps/sig-storage/3314-csi-changed-block-tracking/README.md]: I agree with Ivan, the capability flag would be for a sidecar to determine if the driver supports CBT feature.

The backup client finds out if VolumeSnapshots are supported - by looking at the VolumeSnapshotClass resource. We can follow a similar approach for CBT availability, so by looking at `SnapshotServiceConfiguration` object.

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md (hunk 2 above)
Comment (PrasadG193) [reply keps/sig-storage/3314-csi-changed-block-tracking/README.md]: Should we have a default `expiryTime` if not configured at the manager?

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md (hunk 1 above)
Comment (carlbraganza) [reply keps/sig-storage/3314-csi-changed-block-tracking/README.md]: Thanks. Then there is no need for us have a formal capability mechanism.

File: keps/sig-storage/3314-csi-changed-block-tracking/README.md (hunk 2 above)
Comment (carlbraganza) [reply keps/sig-storage/3314-csi-changed-block-tracking/README.md]: Definitely should have a flag with default that can be overridden by a config map.  We can leave it to the implementation but require such configurability.

**Pull request merged**